src/Pos/Wallet/Web/ClientTypes.hs

```diff
@@ -248,12 +249,22 @@ instance Show CPassPhrase where
 
 type instance OriginType CPassPhrase = PassPhrase
 
+-- These two instances nearly duplicate `instance Bi PassPhrase`
+-- in `Pos.Binary.Crypto`.
 instance FromCType CPassPhrase where
-    decodeCType (CPassPhrase text) =
-        first toText . decodeFull  =<< Base16.decode text
+    decodeCType (CPassPhrase text) = do
+        bs <- Base16.decode text
+        let bl = BS.length bs
+        -- Currently passphrase may be either 32-byte long or empty (for
+        -- unencrypted keys).
+        if bl == 0 || bl == passphraseLength
+            then pure $ ByteArray.convert bs
+            else fail . toString $ sformat
+                 ("Expected password length 0 or "%int%", not "%int)
+                 passphraseLength bl
```

Comment:
I really don't like such copypaste, but since we cannot simply re-use `Binary` instance here, I think I can live with it (until next serialization refactoring though =\)